### ISSUE TITLE
Not able to save recordfields in a custom workflow

### ISSUE THREAD

**Issue by jkulker** [umbraco/Umbraco.Forms.Issues#431]
I created a customwork workflow that is adding values to specific fields. The code hits all the code but not saved in the database.

```
public override WorkflowExecutionStatus Execute(Record record, RecordEventArgs e)
        {
            var correctCount = 0;
            var questionIsTrue = "true";
            foreach (var recordField in record.RecordFields.Values)
            {
                if (recordField.HasValue() && recordField.Values.Any(x => questionIsTrue.Contains(x.ToString())))
                {
                        correctCount++;
                }

                if (recordField.Alias == "passed")
                {
                    recordField.Values.Add(correctCount >= this.correctAnswersNeeded);
                }

                if (recordField.Alias == "correctAnswers")
                {
                    recordField.Values.Add(correctCount);
                }
                record.
            }

            // HttpContext.Current.Response.Redirect("resultaat");
            return WorkflowExecutionStatus.Completed;
        }
```
Umbraco: 8.8
Forms: 8.5.3

### Bug summary

Impossible to save data in recordfield

---
_This item has been added to our backlog [AB#9369](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/9369)_

**Comment by jordanc44** [comment]
having the same issue

**Comment by jordanc44** [comment]
I'm also having the same issue. I have a workflow that runs on submit that changes the value of a hidden field. This worked fine on forms version 8.3 but now none of my changes are saved after upgrading to 8.5.3.

My code:
```
 public override WorkflowExecutionStatus Execute(Record record, RecordEventArgs e)
        {

            // Iterate through the form fields
            foreach (RecordField rf in record.RecordFields.Values)
            {
                // Find the hidden value
                if(rf.Alias.ToLower() == "ordernumber")
                {
                    // set the value to the current order number
                    rf.Values.Clear();
                    rf.Values = new List<object> { _orderService.GetOrderNumber() };
                }
            }
            return WorkflowExecutionStatus.Completed;
        }
```

**Comment by nul800sebastiaan** [comment]
Thanks, I can confirm this worked in Forms 8.3.2 and stopped working in v8.4. We'll need to have a look.

The full code I used:

```csharp
using System;
using System.Collections.Generic;
using System.Linq;
using Umbraco.Forms.Core;
using Umbraco.Forms.Core.Enums;
using Umbraco.Forms.Core.Persistence.Dtos;

namespace MyNamespace
{
  public class CustomWorkflow : WorkflowType
   {
      public override WorkflowExecutionStatus Execute(Record record, RecordEventArgs e)
      {
          foreach (var rf in record.RecordFields.Values.Where(rf => rf.Alias.ToLower() == "ordernumber"))
          {
              // set the value to the current order number
              rf.Values.Clear();
              rf.Values = new List<object> { "overruled" };
          }


          return WorkflowExecutionStatus.Completed;
      }

      public override List<Exception> ValidateSettings()
      {
          return new List<Exception>();
      }
   }
}
```

I have a field of type short answer with the alias of `ordernumber` and it doesn't matter if I try to store forms in the database or not, neither of them accept the value being changed.

**Comment by warrenbuckley** [comment]
Thanks for the example @nul800sebastiaan 
I can reproduce and now playing the lovely game of spot the difference to see what's happened and is causing this issue

https://github.com/umbraco/Forms/compare/v8.3.2...v8.4.0

**Comment by warrenbuckley** [comment]
As I am running out of time before I hit holiday for a few days, but debugging and stepping through the problem seems to lies in the changes in the workflow service. Linking this here for reference when I come back next week

https://github.com/umbraco/Forms/compare/v8.3.2...v8.4.0#diff-ce787f660aac487815cf302630d96b96469225328047398238ddb583078567ae

**Comment by miackers** [comment]
Hi @warrenbuckley, any update on this? 

I am having the same issue running Umbraco Forms v8.5.5, and tried v8.60 now as well. 

Mine is slightly different - I am using code similar to https://github.com/umbraco/Umbraco.Forms.Issues/issues/186#issuecomment-511328665 , biggest difference being I am calling 
`
record.Created = DateTime.Now;
_recordStorage.UpdateRecord(record, e.Form);
`
as well.

Last known version where I know this worked was in Umbraco Forms v8.3.2

Environment details:
- Umbraco v8.6.5
- Umbraco Forms v8.5.5 & v8.60
- StoreUmbracoFormsInDb = False (Using UmbracoFileSystemProviders.Azure.Form v2.0.0 for storage)
- Have load balanced environment, but my local/dev box is set to MasterServer, so I doubt that matters

If you need any more information, let me know.

**Comment by warrenbuckley** [comment]
This has been fixed and will be in the next version 8.6.1
